doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,187 @@
+
+{{< note >}}
+The `PARTITION BY` option has no impact on the order in which records are returned by queries.
+If you want to return results in a specific order, use an `ORDER BY` clause on your [`SELECT` statement](/sql/select/).
+{{< /note >}}
+
+## Syntax
+
+The option `PARTITION BY <column list>` declares that a [materialized view](/sql/create-materialized-view/#with_options) or [table](/sql/create-table/#with_options) should be partitioned by the listed columns.
+For example, a table that stores an append-only collection of events may want to partition the data by time:
+
+```mzsql
+CREATE TABLE events (event_ts timestamptz, body jsonb)
+WITH (
+    PARTITION BY (event_ts)
+);
+```
+
+This `PARTITION BY` clause declares that events with similar `event_ts` timestamps should be stored together.
+
+When multiple columns are specified, rows are partitioned lexicographically.
+For example, `PARTITION BY (event_date, event_time)` would partition first by the created date;
+if many rows have the same `event_date`, those rows would be partitioned by the `event_time` column.
+Durable collections without a `PARTITION BY` option can be partitioned arbitrarily.
+
+{{< note >}}
+The `PARTITION BY` option does not mean that rows with different values for the specified columns will be stored in different parts, only that rows with similar values for those columns should be stored together.
+{{< /note >}}
+
+## Requirements
+
+Materialize currently imposes some restrictions on the list of columns in the `PARTITION BY` clause.
+
+- This clause must list a prefix of the columns in the collection. For example:
+  - if you're creating a table that partitions by a single column, that column must be the first column in the table's schema definition;
+  - if you're creating a table that partitions by two columns, those columns must be the first two columns in the table's schema definition and listed in the same order.
+- Only certain types of columns are supported. This includes:
+    - all fixed-width integer types, including `smallint`, `integer`, and `bigint`;
+    - date and time types, including `date`, `time`, `timestamp`, `timestamptz`, and `mz_timestamp`;
+    - string types like `text` and `bytea`;
+    - `boolean` and `uuid`;
+    - `record` types where all fields types are supported.
+
+We intend to relax some of these restrictions in the future.
+
+## Filter pushdown
+
+Suppose we're running a query against our `events` data with a [temporal filter](/transform-data/patterns/temporal-filters/).
+
+```mzsql
+SELECT * FROM events WHERE mz_now() <= event_ts + INTERVAL '5min';
+```
+
+This query returns only rows with similar values for `event_ts`: timestamps in the last five minutes.
+Since we declared that our `events` table is partitioned by `event_ts`, Materialize can determine that all the rows that pass this filter must also be _stored_ close together, and fetch only the exact parts it needs to satisfy the query.
```

Comment:
I find this explanation a bit unsatisfying.

>and fetch only the exact parts it needs to satisfy the query.

It does not really explain how. That may be on purpose, which would be fine. But if we wanted to share some more details, we could add a sentence or two on how the metadata of parts can be used to prune parts without reading the data and how ordering makes this more effective.

Reply:
I think I can probably say a bit more here without getting _too_ deep in the weeds... I'll take another look.

Reply:
Took a stab at it!

Reply:
Looks great, thanks!